sound/soc/sof/intel/hda-dsp.c

```diff
@@ -574,31 +574,11 @@ static void hda_dsp_state_log(struct snd_sof_dev *sdev)
  * is called again either because of a new IPC sent to the DSP or
  * during system suspend/resume.
  */
-int hda_dsp_set_power_state(struct snd_sof_dev *sdev,
-			    const struct sof_dsp_power_state *target_state)
+static int hda_dsp_set_power_state(struct snd_sof_dev *sdev,
+				   const struct sof_dsp_power_state *target_state)
 {
 	int ret = 0;
 
-	/*
-	 * When the DSP is already in D0I3 and the target state is D0I3,
-	 * it could be the case that the DSP is in D0I3 during S0
-	 * and the system is suspending to S0Ix. Therefore,
-	 * hda_dsp_set_D0_state() must be called to disable trace DMA
-	 * by sending the PM_GATE IPC to the FW.
-	 */
-	if (target_state->substate == SOF_HDA_DSP_PM_D0I3 &&
-	    sdev->system_suspend_target == SOF_SUSPEND_S0IX)
-		goto set_state;
```

Comment:
@ujfalusi if the system suspends to S0ix while a deep buffer pcm is active, it will not be suspended. So the DSP will continue to be D0i3 and then hda_dsp_suspend() will try to set the DSP to d0i3 again.

Reply:
@ranj063, like `aplay -iv -Dplughw:0,31 -fdat /dev/zero` ?
If I do that then the audio stopped prior and the DSP is in D0I0 (tested on TGL), I need WoV or something like that to reproduce or test this?